modelica_builder/modelica_mos_file.py

```diff
@@ -77,23 +103,93 @@ def replace_header_variable_value(self, key: str, new_value: Any) -> bool:
         key_re = rf'#(\s?{key}\s?)=\s?(-?\b\d[\d,.]*\b)(.*\s)'
 
         # verify that group3 exists, otherwise the key doesn't exist
-        match = re.search(key_re, self.data)
+        match = re.search(key_re, self.header_data)
         if match:
             if match.group(3):
                 # if there are units, then put the unit back in
-                self.data = re.sub(key_re, '#' + r'\g<1>' + f"= {new_value}" + r'\g<3>', self.data)
+                self.header_data = re.sub(key_re, '#' + r'\g<1>' + f"= {new_value}" + r'\g<3>', self.header_data)
             else:
                 # no g3, therefore, just replace the var and value.
-                self.data = re.sub(key_re, '#' + r'\g<1>' + f"= {new_value}", self.data)
+                self.header_data = re.sub(key_re, '#' + r'\g<1>' + f"= {new_value}", self.header_data)
 
         # replace the value, this requires recreating the variable and value
-        self.data = re.sub(key_re, '#' + r'\g<1>' + f"= {new_value}" + r'\g<3>', self.data)
+        self.header_data = re.sub(key_re, '#' + r'\g<1>' + f"= {new_value}" + r'\g<3>', self.header_data)
+        return True
+
+    def scale_loads(self, scale_factors: list, load_type: Union[str, int, list[int]] = "all") -> bool:
+        """Scale the loads by a factor. The scale factors must look like the following:
```

Comment:
Can you explain the use case? Why would we scale loads, and what does that do to the file and for the user?